common/core/web/tools/sentry-manager/src/index.ts

```diff
@@ -0,0 +1,137 @@
+    }
+
+    constructor(options: Options = KeymanSentryManager.DEFAULT_OPTIONS) {
+      this.keymanPlatform = options.hostPlatform;
+    }
+
+    // If we've recognized one of our source files, 
+    aliasFilename(filename: string): string|null {
+      if(!this.mayAlias(filename)) {
+        return null;
+      }
+
+      filename = KeymanSentryManager.STANDARD_ALIASABLE_FILES[filename];
+
+      switch(location.protocol) {
+        case 'http:':
+          return 'http://' + location.host + '/' + filename;
+        case 'file:':
+          return 'file:///' + filename;
+        default:
+          return null;
+      }
+    }
+
+    mayAlias(filename: string): boolean {
+      return !!KeymanSentryManager.STANDARD_ALIASABLE_FILES[filename];
+    }
+
+    // Filters all expected but unnecessary path prefixes, affixes, and suffixes reported by Sentry from our products.
+    // This allows us to mask all different sorts of installations with a single uploaded path.
+    pathFilter(event: any) {
+      // Get the underlying JS error.
+      let exception = event.exception;
+
+      // Iterate through all wrapped exceptions.
+      for(let e of exception.values) {
+        for(let frame of e.stacktrace.frames) {
+          let URL = frame.filename as string;
+          let filename: string = '';
+          try {
+            filename = URL.substr(URL.lastIndexOf('/')+1);
+            if(this.mayAlias(filename)) {
+              // Make the file look like it's at the host's root so that the Sentry server will recognize it
+              // regardless of actual filepath position.
+              frame.filename = this.aliasFilename(filename) || frame.filename;
+            }
+          } catch {
+            // not sure what to do yet, but... don't make more errors in the error handler.  That's a bad idea.
+          }
+        }
+      }
+
+      return event;
+    }
+
+    attachEventMetadata(event: any) {
+      // Ensure that the 'extra' object exists.  (May not exist for synthetic/custom Errors.)
+      event.extra = event.extra || {};
+      event.extra.keymanState = window['keyman']['getDebugInfo']();
+      event.extra.keymanHostPlatform = this.keymanPlatform;
+      return event;
+    }
+
+    /**
+     * Pre-processes a Sentry event object to provide more metadata and enhance the Sentry server's
+     * ability to match the error against release artifacts.
+     * @param event A Sentry-generated event
+     */
+    eventPreparer(event: any) {
+      event = this.pathFilter(event);
+      event = this.attachEventMetadata(event);
+
+      if(DEBUG) {
+        console.log("DEBUG:  event object for Sentry")
+        console.log(event);
+        return null; //event
+      } else {
+        return event;
+      }
+    }
+
+    /**
+     * Allows debugging our custom event preparation code without bombarding Sentry with errors
+     * during development.
+     * @param event
+     */
+    __metaPreparer(event: any) {
```

Comment:
I don't understand this naming scheme.

Reply:
What would you prefer, then?  Something like `__prepareEventDebugWrapper`?

Reply:
Why the `__`? `prepareEventDebugWrapper` tells me more about what the function does, so that's definitely an improvement 😄